website/hw/hw7.rst

```diff
@@ -231,65 +250,85 @@ repository:
+    https://github.com/uchicago-cmsc22000/2021-hw7-GITHUB_USERNAME/actions/runs/XXXXXXXX
+
+Now, go back and fix the app so that the tests pass again. Then, push your
 fixed app to both GitHub and to Heroku. Take the URL of the successful
-build, and enter it in Gradescope.
+run, and enter it in Gradescope.
 
-Task 4: Deploy using Github Integration with Travis CI
-======================================================
+Task 4: Deploy using Github Integration
+---------------------------------------
 
 [20 points]
 
 Wouldn’t it be convenient if we could deploy continuously, as soon as
-tests pass? You might think “well, what if we had a deploy phase in our
-Travis CI”? This is possible (see
-https://docs.travis-ci.com/user/deployment/heroku/), but it turns out
+tests pass? You might think “well, what if we had a deploy job in our
+GitHub Actions workflow”? This is possible (see
+https://github.com/marketplace/actions/deploy-to-heroku), but it turns out
 Heroku makes it *even easier* than that. On your Heroku app dashboard,
 in the “Deploy” tab, under “Deployment method”, select “GitHub - connect
 to GitHub” instead of “Heroku Git - Use Heroku CLI”.
 
 Then, you should be able to connect to your repo. To do this, you need
-to first select the ``cmsc22000-labs`` organization under “Search for a
+to first select the ``uchicago-cmsc22000`` organization under “Search for a
 repository to connect to”, and then enter your repository’s name
-(``2020-lab7-GITHUB_USERNAME``) in the text field (make sure to click
-the “Search” button so Heroku will search for your repository).
+(``2021-hw7-GITHUB_USERNAME``) in the text field. Make sure to click
+the “Search” button so Heroku will search for your repository; your
+repository should then appear under the search field, with a "Connect"
+button next to it. Click on that button to connect your repository.
 
-Once you connect the repo, you should see an option that says “Automatic
+Once you connect the repo, you should see a new section called “Automatic
 deploys” with a checkbox “Wait for CI to pass before deploy”. Make sure
 that checkbox is checked, and then click on “Enable Automatic Deploys”
 
 Make the tests fail again and push to GitHub with a simple ``git push``.
 The CI tests should fail and, if you navigate to your URL, you’ll see
 that the broken version has not been deployed.
 
-{{% warning %}} **Caution:** In the above ``git push`` step, *don’t*
-push to Heroku as you did in previous tasks. What’s happening here is
-that the Heroku service will now wait for tests to pass CI, and then
-automagically pull your code and deploy it. No need to manually push to
-Heroku. {{% /warning %}}
+.. warning::
+
+    **Caution:** In the above ``git push`` step, *don’t*
+    push to Heroku as you did in previous tasks. What’s happening here is
+    that the Heroku service will now wait for tests to pass CI, and then
+    automagically pull your code and deploy it. No need to manually push to
+    Heroku.
```

Comment:
At the end of Task 3 we are telling students to fix the code and push to both GitHub and Heroku, but this statement in a warning box seems to contradict that (unless I'm understanding this incorrectly). I think we should tell them to not fix and/or push their code at the end of Task 3 and have this be the only location where they do that.

Reply:
I've modified that part of the task to make it clearer what they have to do, and *why* we're asking them to do things in that order.

fyi, the intended flow is the following:

- At the end of Task 3, they're supposed to leave both their repo and their deployed app in a correct state. 
- Then, in Task 4, we setup auto-deploys, and ask them to intentionally break their app and push it to GitHub (but *not* Heroku) to demonstrate how Heroku won't deploy an app that doesn't pass CI. 
- Then, we ask them to fix the app and make a change that doesn't break the tests, to demonstrate that Heroku will deploy that app as long as it passes the CI tests. 

The warning box was meant to convey that they shouldn't run `git push heroku main` because that defeats the purpose of using automatic deployments (I've made this more explicit)